src/main/java/inu/parser/Parser.java

```diff
@@ -0,0 +1,216 @@
+package inu.parser;
+
+import inu.exceptionhandling.EmptyTaskListException;
+import inu.exceptionhandling.EmptyUserInputException;
+import inu.exceptionhandling.ExceptionManager;
+import inu.task.DeadLine;
+import inu.task.Event;
+import inu.task.Task;
+import inu.task.Todo;
+import inu.task.TaskList;
+import inu.commons.Ui;
+import inu.commons.Util;
+
+
+public class Parser {
+
+    protected static final int DECODED_TASK_TYPE = 0;
+
+    protected static final int DECODED_MARK = 1;
+
+    protected static final int DECODED_TASK = 2;
+
+    protected static final int DECODED_DUE_DATE = 3;
+
+    protected static final int DECODED_FROM_DATE = 3;
+
+    protected static final int DECODED_TO_DATE = 4;
+
+    protected static final String T = "T";
+
+    protected static final String D = "D";
+
+    protected static final String E = "E";
+
+    protected static final String MARKED = "X";
+
+    public static void runTodo(TaskList taskList, String entry) {
+
+        try {
+
+            ExceptionManager.checkEmptyUserInput(entry);
+            Todo todoTask = new Todo(entry);
+            taskList.addTask(todoTask);
+            Ui.printAcknowledgment(taskList);
+
+        } catch (EmptyUserInputException e) {
+
+            Ui.printPromptValidTaskEntry();
+
+        }
+
+    }
+
+    public static void runDeadLine(TaskList taskList, String entry) {
+
+        try {
+
+            String task = Util.fetchTask(entry);
+            String deadLine = Util.fetchDeadLine(entry);
+            ExceptionManager.checkEmptyUserInput(task, deadLine);
+            DeadLine deadLineTask = new DeadLine(task, deadLine);
+            taskList.addTask(deadLineTask);
+            Ui.printAcknowledgment(taskList);
+
+        } catch (EmptyUserInputException e) {
+
+            Ui.printPromptValidByDateEntryAfterSlash();
+
+        } catch (StringIndexOutOfBoundsException e) {
+
+            Ui.printPromptValidDeadLine();
+
+        }
+
+    }
+
+    public static void runEvent(TaskList taskList, String entry) {
+
+        try {
+
+            String task = Util.fetchTask(entry);
+            String from = Util.fetchFrom(entry);
+            String to = Util.fetchTo(entry);
+            ExceptionManager.checkEmptyUserInput(task, from, to);
+            Event eventTask = new Event(task, from, to);
+            taskList.addTask(eventTask);
+            Ui.printAcknowledgment(taskList);
+
+        } catch (EmptyUserInputException e) {
+
+            Ui.printPromptValidFromAndToEntryAfterSlash();
```

Comment:
Avoid using method or variable names which are too long and difficult to read/understand. Be specific but succinct :)